torchvision/transforms/functional.py

```diff
@@ -329,6 +329,8 @@ def pad(img, padding, fill=0, padding_mode='constant'):
         'Padding mode should be either constant, edge, reflect or symmetric'
 
     if padding_mode == 'constant':
+        if isinstance(fill, numbers.Number):
+            fill = (fill,) * len(img.getbands())
```

Comment:
We should also check if `len(img.getbands())` and `len(fill)` match if `fill` is not an `int`.

Reply:
Good point, added another check